pkg/webhooks/mutator/hyperConvergedMutator.go

```diff
@@ -106,20 +107,42 @@ func (hcm *HyperConvergedMutator) mutateHyperConverged(_ context.Context, req ad
 	return admission.Allowed("")
 }
 
-func mutateEvictionStrategy(hc *hcov1beta1.HyperConverged, patches []jsonpatch.JsonPatchOperation) []jsonpatch.JsonPatchOperation {
+func mutateEvictionStrategy(ctx context.Context, cli client.Client, hc *hcov1beta1.HyperConverged, patches []jsonpatch.JsonPatchOperation) []jsonpatch.JsonPatchOperation {
 	if hc.Status.InfrastructureHighlyAvailable == nil || hc.Spec.EvictionStrategy != nil { // New HyperConverged CR
 		return patches
 	}
 
-	var value = kubevirtcorev1.EvictionStrategyNone
-	if *hc.Status.InfrastructureHighlyAvailable {
-		value = kubevirtcorev1.EvictionStrategyLiveMigrate
+	workerNodes := &corev1.NodeList{}
+	err := cli.List(ctx, workerNodes, client.MatchingLabels{"node-role.kubernetes.io/worker": ""})
+	if err != nil {
+		hcMutatorLogger.Error(err, "Failed to list worker nodes")
+		return patches
+	}
+
+	allArm64 := true
+	for _, node := range workerNodes.Items {
+		arch, found := node.Labels["kubernetes.io/arch"]
+		if !found || arch != "arm64" {
+			allArm64 = false
+			break
+		}
+	}
+
+	var evictionStrategy kubevirtcorev1.EvictionStrategy
+
+	switch {
+	case allArm64:
+		evictionStrategy = kubevirtcorev1.EvictionStrategyNone
+	case hc.Status.InfrastructureHighlyAvailable != nil && *hc.Status.InfrastructureHighlyAvailable:
+		evictionStrategy = kubevirtcorev1.EvictionStrategyLiveMigrate
+	default:
+		evictionStrategy = kubevirtcorev1.EvictionStrategyNone
```

Comment:
nit (not sure it's better): we only have two cases here. I wonder if this we;ll be more clear:
```suggestion
    evictionStrategy := kubevirtcorev1.EvictionStrategyNone
    if !allArm64 && hc.Status.InfrastructureHighlyAvailable != nil && *hc.Status.InfrastructureHighlyAvailable {
       evictionStrategy = kubevirtcorev1.EvictionStrategyLiveMigrate
    }
```

Reply:
Thanks, it's quite nice actually.